File: 00404. Sum of Left Leaves/00404.go
// https://leetcode.com/problems/sum-of-left-leaves

func helper(node *TreeNode, isLeft bool, sum *int) {
    if node == nil {
        return
    }

    if isLeft && node.Left == nil && node.Right == nil {
        *sum += node.Val
        return
    }

    helper(node.Left, true, sum)
    helper(node.Right, false, sum)
}

func sumOfLeftLeaves(root *TreeNode) (sum int) {
    helper(root, false, &sum)
    return sum
}

File: LeetCode/00404. Sum of Left Leaves/00404.c
// https://leetcode.com/problems/sum-of-left-leaves

void helper(struct TreeNode *node, bool isLeft, int *sum) {
    if (node == NULL) {
        return;
    }

    if (isLeft && node->left == NULL && node->right == NULL) {
        *sum += node->val;
        return;
    }

    helper(node->left, true, sum);
    helper(node->right, false, sum);
}

int sumOfLeftLeaves(struct TreeNode *root) {
    int sum = 0;
    helper(root, false, &sum);
    return sum;
}
